RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronCoreProducer.cc

```diff
@@ -1,62 +1,79 @@
+};
+
+using reco::GsfPFRecTrackCollection;
+using reco::SuperClusterRef;
 
 LowPtGsfElectronCoreProducer::LowPtGsfElectronCoreProducer(const edm::ParameterSet& config)
-    : GsfElectronCoreBaseProducer(config) {
-  superClusterRefs_ =
-      consumes<edm::ValueMap<reco::SuperClusterRef> >(config.getParameter<edm::InputTag>("superClusters"));
-}
+    : gsfPfRecTracksToken_(mayConsume<GsfPFRecTrackCollection>(config.getParameter<edm::InputTag>("gsfPfRecTracks"))),
+      ctfTracksToken_(consumes<reco::TrackCollection>(config.getParameter<edm::InputTag>("ctfTracks"))),
+      superClusterRefs_(consumes<edm::ValueMap<SuperClusterRef> >(config.getParameter<edm::InputTag>("superClusters"))),
+      putToken_(produces<reco::GsfElectronCoreCollection>()) {}
 
-void LowPtGsfElectronCoreProducer::produce(edm::Event& event, const edm::EventSetup& setup) {
+void LowPtGsfElectronCoreProducer::produce(edm::StreamID, edm::Event& event, const edm::EventSetup& setup) const {
   // Output collection
-  auto electrons = std::make_unique<reco::GsfElectronCoreCollection>();
+  reco::GsfElectronCoreCollection electrons;
 
   // Init
-  GsfElectronCoreBaseProducer::initEvent(event, setup);
-  if (!useGsfPfRecTracks_) {
-    edm::LogError("useGsfPfRecTracks_ is (redundantly) set to False!");
-  }
-
-  edm::Handle<edm::ValueMap<reco::SuperClusterRef> > superClusterRefs;
-  event.getByToken(superClusterRefs_, superClusterRefs);
+  auto gsfPfRecTracksHandle = event.getHandle(gsfPfRecTracksToken_);
+  auto ctfTracksHandle = event.getHandle(ctfTracksToken_);
+  auto const& superClusterRefs = event.get(superClusterRefs_);
 
   // Create ElectronCore objects
-  for (size_t ipfgsf = 0; ipfgsf < gsfPfRecTracksH_->size(); ++ipfgsf) {
+  for (size_t ipfgsf = 0; ipfgsf < gsfPfRecTracksHandle->size(); ++ipfgsf) {
     // Refs to GSF(PF) objects and SC
-    reco::GsfPFRecTrackRef pfgsf(gsfPfRecTracksH_, ipfgsf);
-    reco::GsfTrackRef gsf = pfgsf->gsfTrackRef();
-    const reco::SuperClusterRef sc = (*superClusterRefs)[pfgsf];
+    reco::GsfPFRecTrackRef pfgsf(gsfPfRecTracksHandle, ipfgsf);
 
     // Use GsfElectronCore(gsf) constructor and store object via emplace
-    electrons->emplace_back(gsf);
+    electrons.emplace_back(pfgsf->gsfTrackRef());
 
     // Do not consider ECAL-driven objects
-    if (electrons->back().ecalDrivenSeed()) {
-      electrons->pop_back();
+    if (electrons.back().ecalDrivenSeed()) {
+      electrons.pop_back();
       continue;
     }
 
     // Add GSF(PF) track information
-    GsfElectronCoreBaseProducer::fillElectronCore(&(electrons->back()));
+    auto ctfpair = egamma::getClosestCtfToGsf(electrons.back().gsfTrack(), ctfTracksHandle);
+    electrons.back().setCtfTrack(ctfpair.first, ctfpair.second);
 
     // Add super cluster information
-    electrons->back().setSuperCluster(sc);
+    electrons.back().setSuperCluster(superClusterRefs[pfgsf]);
   }
 
-  event.put(std::move(electrons));
+  event.emplace(putToken_, electrons);
```

Comment:
Just to note that this will call the copy constructor of `reco::GsfElectronCoreCollection` inside `emplace()`. Using `move` like
```cpp
event.emplace(putToken_, std::move(electrons));
```
would use the move constructor (that would be much cheaper for e.g. `std::vector`s).

Reply:
Thanks @makortel for the heads up and taking care! I'm changing that of course.